Classes/YLTableViewDataSource.m

```diff
@@ -14,12 +14,28 @@
 #import "YLTableViewPrivate.h"
 #import "YLTableViewCell.h"
 #import "YLTableViewCellPrivate.h"
+#import "YLTableViewCellEstimatedRowHeight.h"
 #import "YLTableViewChildViewControllerCell.h"
 #import "YLTableViewSectionHeaderFooterView.h"
 #import "YLTableViewSectionHeaderFooterViewPrivate.h"
 
+@interface YLTableViewDataSource ()
+
+//! This is used to cache the estimated row heights
+@property (strong, nonatomic) NSMutableDictionary * /*of NSString*/ indexPathToEstimatedRowHeight;
+
+@end
+
 @implementation YLTableViewDataSource
 
+
+-(instancetype)init {
+  if (self = [super init]) {
+    _indexPathToEstimatedRowHeight = [[NSMutableDictionary alloc] init];
```

Comment:
since this is opt-in, maybe we should only setup this dictionary if the client has opted in